src/i2c.rs

```diff
@@ -19,16 +19,38 @@ crate::embedded_hal_error!(
     embedded_hal::i2c::ErrorKind
 );
 
+#[cfg(any(esp32, esp32s2))]
 const APB_TICK_PERIOD_NS: u32 = 1_000_000_000 / 80_000_000;
+#[cfg(not(any(esp32, esp32s2)))]
+const XTAL_TICK_PERIOD_NS: u32 = 1_000_000_000 / XTAL_CLK_FREQ;
 #[derive(Copy, Clone, Debug)]
 pub struct APBTickType(::core::ffi::c_int);
 impl From<Duration> for APBTickType {
+    #[cfg(any(esp32, esp32s2))]
     fn from(duration: Duration) -> Self {
         APBTickType(
             ((duration.as_nanos() + APB_TICK_PERIOD_NS as u128 - 1) / APB_TICK_PERIOD_NS as u128)
                 as ::core::ffi::c_int,
         )
     }
+    #[cfg(not(any(esp32, esp32s2)))]
+    /// Conversion for newer esp models, be aware, that the hardware can only represent 22 different values, values will be rounded to the next larger valid one. Calculation only valid for 40mhz clock source
+    fn from(duration: Duration) -> Self {
+        let target_ns = duration.as_nanos();
+        let target_ns_f = target_ns as u64;
+        let abc = target_ns_f / (XTAL_TICK_PERIOD_NS as u64);
+        let i = abc.ilog2()
+            + (if abc.leading_zeros() + abc.trailing_zeros() + 1 < 64 {
+                1
+            } else {
+                0
+            });
+        if i <= 22 {
```

Comment:
Doesn't esp idf define a constant already for `22`? If yes, can you use it?

Reply:
There seems to exist I2C_LL_MAX_TIMEOUT, at least the arduino hal seems to use that, but it seems to not be included in the current bindings, how would I add this?